content/articles/the-complete-roadmap-to-blockchain-development/index.md

```diff
@@ -0,0 +1,116 @@
+
+ In the public testnets and the mainnet, you would need to make use of a service called Infura. Infura is an API that runs ethereum clients for you since it is not easy to run clients on your own. In addition to that, the way you send transactions in Ganache is quite different compared to how transactions take place in public testnets and mainnet. The learning curve increases exponentially with practice.
+
+ Finally, you need to test your smart contract appropriately before deploying it since as I mentioned earlier, the code is immutable. The testing is easily achieved using the truffle framework. 
+
+ I hope you can see that smart contracts cover a large part of blockchain development but I also hope that you can appreciate that it is not wise to learn smart contracts in isolation.
+
+  #### 4. Front-End Development
+
+ Now you have successfully deployed your smart contract on the blockchain. Problem is that the only way to interact with it is using the command line. The command line sends shivers down the spine of most people even with a technical background. You wouldn't want that for your users, right? that's why you need to create a friendly user interface at the front end. The front-end interface you create is called a DAPP(Decentralized application). The DAPP can be a mobile app or a web app but in most cases, it is usually a web app. The web app is usually just like your normal web apps with HTML, CSS, and javascript. This further emphasizes my point that you need to understand web development before you get started with blockchain app development.
+
+ In the design of the DAPP, there will be two main challenges; first, the integration with the blockchain, and secondly the integration with the wallet. For integration with the blockchain, we shall make use of a javascript library called Web3 Js which is quite easy and help can be found online. Integration with the wallet is a little bit sophisticated since in DAPPs the user management is decentralized and users store "passwords" themselves. Actually, in DAPPS we don't usually use passwords but the closest equivalent is what we call private keys. With private keys, the users can sign a transaction, which is a data package that describes an action that a user wants to do, with a verifiable signature that proves the user wanted to do that transaction. The wallet I recommend for integrating with the ethereum blockchain is Metamask, though there are a lot of crypto wallets out there. So you should learn first how to integrate your smart contract with the Metamask wallet and then learn for other wallets.
```

Comment:
```suggestion
In the design of the DAPP, there will be two main challenges; first, the integration with the blockchain, and secondly the integration with the wallet. For integration with the blockchain, we shall make use of a JavaScript library called Web3.js, which is quite easy and help can be found online. Integration with the wallet is a little bit sophisticated since in DAPPs the user management is decentralized and users store "passwords" themselves. In DAPPs, we don't usually use passwords but the closest equivalent is what we call private keys. With private keys, the users can sign a transaction, which is a data package that describes an action that a user wants to do, with a verifiable signature that proves the user wanted to do that transaction. The wallet recommended for integrating with the ethereum blockchain is Metamask, though there are a lot of crypto wallets out there. You should first learn how to integrate your smart contract with the Metamask wallet and then learn the other wallets.